src/software/jetson_nano/services/motor.cpp

```diff
@@ -119,13 +123,43 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
 MotorService::~MotorService() {}
 
 std::unique_ptr<TbotsProto::DriveUnitStatus> MotorService::poll(
-    const TbotsProto::DirectControlPrimitive_DirectVelocityControl& local_velocity,
-    float dribbler_speed_rpm)
+    const TbotsProto::DirectControlPrimitive& direct_control)
 {
-    // TODO (#2335) convert local velocity to per-wheel velocity
-    // using http://robocup.mi.fu-berlin.de/buch/omnidrive.pdf and then
-    // communicate velocities to trinamic. Also read back feedback and
-    // return drive unit status.
+    // TODO (#2335) We can only spin 1 motor right now
+    CHECK(encoder_calibrated_[FRONT_LEFT_MOTOR_CHIP_SELECT])
+        << "Running without encoder calibration can cause serious harm";
+
+    switch (direct_control.wheel_control_case())
+    {
+        case TbotsProto::DirectControlPrimitive::WheelControlCase::kDirectPerWheelControl:
+        {
+            // TODO (#2456) until we figure out the right factor considering
+            // the gear ratio
+            static float RANDOM_SCALING_FACTOR = 100.0;
+
+            // TODO (#2335) We can only spin 1 motor right now, figure out the
+            tmc4671_setTargetVelocity(
```

Comment:
incomplete comment?

Reply:
also i think issue 2335 will be resolved after this pr (as per your pr description)

Reply:
good catch, fixed